tools/src/test/java/org/apache/kafka/tools/FeatureCommandTest.java

```diff
@@ -0,0 +1,292 @@
+        String commandOutput = ToolsTestUtils.captureStandardOut(() ->
+                assertEquals(1, FeatureCommand.mainNoExit("--bootstrap-server", cluster.bootstrapServers(),
+                        "upgrade", "--metadata", "3.3-IV2"))
+        );
+        assertEquals("Could not upgrade metadata.version to 6. Could not apply finalized feature " +
+                "update because the provided feature is not supported.", commandOutput);
+    }
+
+    @ClusterTest(clusterType = Type.KRAFT, metadataVersion = MetadataVersion.IBP_3_3_IV1)
+    public void testUpgradeMetadataVersionWithKraft() {
+        String commandOutput = ToolsTestUtils.captureStandardOut(() ->
+                assertEquals(0, FeatureCommand.mainNoExit("--bootstrap-server", cluster.bootstrapServers(),
+                        "upgrade", "--feature", "metadata.version=5"))
+        );
+        assertEquals("metadata.version was upgraded to 5.", commandOutput);
+
+        commandOutput = ToolsTestUtils.captureStandardOut(() ->
+                assertEquals(0, FeatureCommand.mainNoExit("--bootstrap-server", cluster.bootstrapServers(),
+                        "upgrade", "--metadata", "3.3-IV2"))
+        );
+        assertEquals("metadata.version was upgraded to 6.", commandOutput);
+    }
+
+    @ClusterTest(clusterType = Type.ZK, metadataVersion = MetadataVersion.IBP_3_3_IV1)
+    public void testDowngradeMetadataVersionWithZk() {
+        String commandOutput = ToolsTestUtils.captureStandardOut(() ->
+                assertEquals(1, FeatureCommand.mainNoExit("--bootstrap-server", cluster.bootstrapServers(),
+                        "disable", "--feature", "metadata.version"))
+        );
+        assertEquals("Could not disable metadata.version. Can not delete non-existing finalized feature.", commandOutput);
+
+        commandOutput = ToolsTestUtils.captureStandardOut(() ->
+                assertEquals(1, FeatureCommand.mainNoExit("--bootstrap-server", cluster.bootstrapServers(),
+                        "downgrade", "--metadata", "3.3-IV0"))
+        );
+        assertEquals("Could not downgrade metadata.version to 4. Could not apply finalized feature " +
+                        "update because the provided feature is not supported.", commandOutput);
+
+        commandOutput = ToolsTestUtils.captureStandardOut(() ->
+                assertEquals(1, FeatureCommand.mainNoExit("--bootstrap-server", cluster.bootstrapServers(),
+                        "downgrade", "--unsafe", "--metadata", "3.3-IV0"))
+        );
+        assertEquals("Could not downgrade metadata.version to 4. Could not apply finalized feature " +
+                "update because the provided feature is not supported.", commandOutput);
+    }
+
+    @ClusterTest(clusterType = Type.KRAFT, metadataVersion = MetadataVersion.IBP_3_3_IV1)
+    public void testDowngradeMetadataVersionWithKRaft() {
+        String commandOutput = ToolsTestUtils.captureStandardOut(() ->
```

Comment:
Not related to your PR, but I found the stream resource in `captureStandardOut` is not closed, causing resource leaking. Could you fix it in this PR or create another PR to fix it? Thanks.